plugins/orchestrator-backend/src/helpers/queryBuilder.ts

```diff
@@ -0,0 +1,37 @@
+import { Pagination } from '../types/pagination';
+
+export function buildGraphQlQuery(
+  type: string,
+  queryBody: string,
+  whereClause?: string,
+  pagination?: Pagination,
+): string {
+  let query = `{${type}`;
+
+  if (whereClause || pagination) {
+    query += ` (`;
+
+    if (whereClause) {
+      query += `where: {${whereClause}}`;
+      if (pagination) {
+        query += `, `;
+      }
+    }
+    if (pagination) {
+      if (pagination.sortField) {
+        query += `orderBy: {${
+          pagination.sortField
+        }: ${pagination.order?.toUpperCase()}}, `;
+      }
+      query += `pagination: {limit: ${pagination.limit} , offset: ${pagination.offset}}`;
+    }
+
+    query += `) `;
+  }
+  query += ` {`;
+  query += queryBody;
+  query += ` }`;
+  query += ` }`;
```

Comment:
Any reason for not simplifying this code in one line?

Reply:
[9d31097](https://github.com/janus-idp/backstage-plugins/pull/1164/commits/9d310979c478244f6333c6dffbed135f486a48b1)